src/System.Management.Automation/engine/Utils.cs

```diff
@@ -1719,6 +1574,39 @@ internal static Process Start(ProcessStartInfo startInfo, ProcessWindowStyle win
             return processToReturn;
         }
 
+        private static void InvokeShellExecuteEx(object param)
+        {
+            NativeMethods.ShellExecuteInfo shellExecuteInfo = (NativeMethods.ShellExecuteInfo)param;
+            if (!NativeMethods.ShellExecuteEx(shellExecuteInfo))
+            {
+                int errorCode = Marshal.GetLastWin32Error();
+                if (errorCode == 0)
```

Comment:
The error handling code needs to happen in the main thread, otherwise, the Win32Exception thrown from here will be ignored.